x-pack/plugins/enterprise_search/public/applications/enterprise_search_content/components/search_index/index_mappings.tsx

```diff
@@ -5,8 +5,69 @@
  * 2.0.
  */
 
-import React from 'react';
+import React, { useEffect } from 'react';
+
+import { useActions, useValues } from 'kea';
+
+import {
+  EuiCodeBlock,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiLink,
+  EuiPanel,
+  EuiText,
+  EuiTitle,
+} from '@elastic/eui';
+
+import { i18n } from '@kbn/i18n';
+
+import { docLinks } from '../../../shared/doc_links';
+
+import { DocumentsLogic } from './documents_logic';
+import { IndexNameLogic } from './index_name_logic';
 
 export const SearchIndexIndexMappings: React.FC = () => {
-  return <>Index Mappings</>;
+  const { indexName } = useValues(IndexNameLogic);
+  const { makeMappingRequest } = useActions(DocumentsLogic);
+  const { mappingData } = useValues(DocumentsLogic);
+
+  useEffect(() => {
+    makeMappingRequest({ indexName });
+  }, [indexName]);
+
+  return (
+    <EuiPanel>
+      <EuiFlexGroup>
+        <EuiFlexItem grow={2}>
+          <EuiCodeBlock language="json" isCopyable>
+            {JSON.stringify(mappingData, null, 2)}
+          </EuiCodeBlock>
+        </EuiFlexItem>
+        <EuiFlexItem grow={1}>
+          <EuiPanel grow={false} hasShadow={false} hasBorder>
+            <EuiTitle>
+              <h3>
+                {i18n.translate('xpack.enterpriseSearch.content.searchIndex.mappings.title', {
+                  defaultMessage: 'About index mappings',
+                })}
+              </h3>
+            </EuiTitle>
+            <EuiText>
+              <p>
+                {i18n.translate('xpack.enterpriseSearch.content.searchIndex.mappings.description', {
+                  defaultMessage:
+                    'Index mappings tell all the bits of bacon where arrange themselves in the omlet. Perfect bacon arrangement leads to the perfect omlet.',
```

Comment:
We need a copy pass on this, but that doesn't have to be in this PR.

Reply:
Another PR is way to go as we need feedback.